comms/storage/transcode/jobs.go

```diff
@@ -57,30 +73,25 @@ type JobsManager struct {
 	mu sync.RWMutex
 }
 
-func NewJobsManager(jsc nats.JetStreamContext, prefix string, replicaCount int) (*JobsManager, error) {
+const (
+	KvSuffix    = "_kv"
+	objStoreTtl = time.Hour * 24
+)
 
-	// work queue
-	queueName := prefix + "_queue"
-	_, err := jsc.AddStream(&nats.StreamConfig{
-		Name:     queueName,
-		Subjects: []string{queueName},
+func NewJobsManager(jsc nats.JetStreamContext, prefix string, replicaCount int) (*JobsManager, error) {
+	// kv
+	kvBucketName := prefix + KvSuffix
+	kv, err := jsc.CreateKeyValue(&nats.KeyValueConfig{
+		Bucket:   kvBucketName,
 		Replicas: replicaCount,
 	})
 	if err != nil {
 		return nil, err
 	}
 
-	// work subscription
-	workSubscription, err := jsc.QueueSubscribeSync(queueName, queueName, nats.AckWait(time.Minute))
-	if err != nil {
-		return nil, err
-	}
-
-	// kv
-	kv, err := jsc.CreateKeyValue(&nats.KeyValueConfig{
-		Bucket:   prefix + "_kv",
-		Replicas: replicaCount,
-	})
+	// work subscription using kv's underlying stream
+	kvStreamName := "KV_" + kvBucketName
+	workSubscription, err := jsc.QueueSubscribeSync("", kvStreamName, nats.AckWait(time.Minute), nats.BindStream(kvStreamName))
```

Comment:
why is subject an empty string here?

Reply:
empty string defaults to all subjects bound to the stream, and since there's no more subject we're publishing to, this makes sure it catches all the messages. I tested logging out what the consumer is reading, and it's receiving what we'd expect (no extraneous KV-related messages). example subject message it received: `"$KV.0_kv.cldkwwu0900059apate4aeguz"`